lib/chef/knife/azure_server_create.rb

```diff
@@ -374,7 +374,7 @@ def get_extension_status()
                   extension_status[:status] = :extension_status_not_detected
                 end
               # This fix is for linux waagent issue: api unable to deserialize the waagent status.
-              elsif role.at_css('GuestAgentStatus Status').text == 'NotReady' and waagent_status_msg == lnx_waagent_fail_msg
+              elsif role.at_css('GuestAgentStatus Status').text == 'NotReady' && waagent_status_msg == lnx_waagent_fail_msg
```

Comment:
adding parenthesis here would be helpful for clarity, i.e.

elsif (role.at_css('GuestAgentStatus Status').text == 'NotReady') && (waagent_status_msg == lnx_waagent_fail_msg)